File: pa6/Arithmetic.py
#------------------------------------------------------------------------------
#
#  Arithmetic.py
#  Produces the same output (given the same input) as Arithmetic.cpp for 
#  CSE 101 pa6.  Run this program by doing
#
#     python3 Arithmetic.py <input file> <output file>
#
#------------------------------------------------------------------------------
import sys

# check number of command line arguments
if len(sys.argv)!=3:
   print("Usage: "+sys.argv[0]+" <input file> <output file>")
   quit()
# end if

# open files
f1 = open(sys.argv[1], 'r')
f2 = open(sys.argv[2], 'w')

# read from f1
A = int( f1.readline() ) # line 1
f1.readline()            # line 2
B = int( f1.readline() ) # line 3

# write lines to f2
f2.write(str(A)+'\n\n')
f2.write(str(B)+'\n\n')
f2.write(str(A+B)+'\n\n')
f2.write(str(A-B)+'\n\n')
f2.write(str(A-A)+'\n\n')
f2.write(str(3*A-2*B)+'\n\n')
f2.write(str(A*B)+'\n\n')
f2.write(str(A**2)+'\n\n')
f2.write(str(B**2)+'\n\n')
f2.write(str(9*A**4+16*B**5)+'\n\n')

# close files
f1.close()
f2.close()

File: pa6/Arithmetic.cpp
#include<iostream>
#include<fstream>
#include<string>
#include"BigInteger.h"
using namespace std;

#define MAX_LEN 300

int main(int argc, char * argv[]){
    //=============================================================================
    // Step1: Check argument, open output\input files.
    //=============================================================================
    ifstream in;
    ofstream out;

    // check command line for correct number of arguments
    if( argc != 3 ){
        cerr << "Usage: " << argv[0] << " <input file> <output file>" << endl;
        return(EXIT_FAILURE);
    }

    // open files for reading and writing 
    in.open(argv[1]);
    if( !in.is_open() ){
        cerr << "Unable to open file " << argv[1] << " for reading" << endl;
        return(EXIT_FAILURE);
    }
    out.open(argv[2]);
    if( !out.is_open() ){
        cerr << "Unable to open file " << argv[2] << " for writing" << endl;
        return(EXIT_FAILURE);
    }

    // ============================================================================= //
    // Step2: Read BigInteger A and BigInteger B.
    // ============================================================================= //
    string line;
    getline(in, line);
    BigInteger A(line);
    getline(in, line);
    getline(in, line);
    BigInteger B(line);

    normalize(A);
    normalize(B);

    // ============================================================================= //
    // Step3: Perform arithmetic operation.
    // ============================================================================= //
    out << A << endl << endl;
    out << B << endl << endl;
    out << A + B << endl << endl;
    out << A - B << endl << endl;
    out << A - A << endl << endl;
    out << A.mult(BigInteger("3")) - B.mult(BigInteger("2")) << endl << endl;
    out << A * B << endl << endl;
    out << A * A << endl << endl;
    out << B * B << endl << endl;
    out << (A * A * A * A).mult(BigInteger("9")) + (B * B * B * B * B).mult(BigInteger("16")) << endl << endl;

    // ============================================================================= //
    // Step3: Close opened files.
    // ============================================================================= //
    // close files 
    in.close();
    out.close();

    return(EXIT_SUCCESS);
}
